File: fixtures/scale/java/large/src/cryptobench/ciphers/feistel/FeistelKey.java
package cryptobench.ciphers.feistel;

public class FeistelKey {
    private final long master;

    public FeistelKey(long master) {
        this.master = master;
    }

    public int subKey(int round) {
        long mixed = master ^ (0x9E3779B97F4A7C15L * (round + 1));
        return (int) (mixed ^ (mixed >>> 32));
    }

    public static FeistelKey defaultKey() {
        return new FeistelKey(0x0F1E2D3C4B5A6978L);
    }
}

File: fixtures/scale/dart/large/lib/ciphers/feistel/feistel_key.dart
class FeistelKey {
  final int master;

  FeistelKey(this.master);

  int subKey(int round) {
    final mixed = master ^ (0x1E3779B9 * (round + 1));
    return (mixed ^ (mixed >> 16)) & 0xFFFFFFFF;
  }

  factory FeistelKey.defaultKey() => FeistelKey(0x0F1E2D3C);
}
